primitives/src/parachain.rs

```diff
@@ -608,6 +608,15 @@ pub enum ValidityAttestation {
 	Explicit(ValidatorSignature),
 }
 
+/// A type returned by runtime with current session index and a parent hash.
+#[derive(Clone, Eq, PartialEq, Decode, Encode, RuntimeDebug)]
+pub struct SigningContext<Hash> {
+	/// Current session index.
+	pub session_index: sp_staking::SessionIndex,
+	/// Hash of the parent.
+	pub parent_hash: Hash,
```

Comment:
The approach we typically take with the `parachain` primitives is to pin directly to the `crate::Hash` type as opposed to being generic.

Reply:
But when in runtime we call `System::parent_hash()` we will end up with some associated type that is not necessarily convertible to `crate::Hash`.

Reply:
I had the same problem with `BlockNumber` in #918 and worked around it with a `Convert` type in the trait, but I agree it is a bit of a hack.